torch/nn/modules/module.py

```diff
@@ -1318,7 +1318,7 @@ def _maybe_warn_non_full_backward_hook(self, inputs, result, grad_fn):
                               "behavior.")
 
     def register_forward_pre_hook(
-        self, hook: Callable[..., None], prepend: bool = False
+        self, hook: Callable[..., None], prepend: bool = False, with_kwargs: bool = False
```

Comment:
Can we make this kwarg only btw?

Reply:
yep that make sense 👍

Reply:
I think the intention was to change the hook signature from
```
def register_forward_pre_hook(
    self, hook: Callable[..., None], prepend: bool = False, with_kwargs: bool = False
)
```
to
```
def register_forward_pre_hook(
    self, hook: Callable[..., None], *, prepend: bool = False, with_kwargs: bool = False
)
```
(`prepend` was added before this PR, but we also want `prepend` to be a keyword-only argument. Since this was added very recently, it should be safe to make this change.)

We can keep the argument itself to be called `with_kwargs: bool` instead of `kwargs: bool`.